releng/tools/rmake

```diff
@@ -30,45 +30,46 @@ usage="""Usage:
 
     rmake clean                  Clean all suite builds
     rmake clean-<suitename>...   Clean given suite
+
+    Use -jN to set N max threads to use during compilation; default is maximum cores
 """
 
-def invokeBuild(wd, targets):
+def invokeBuild(wd, targets, maxThreads):
     """
     Invoke ninja or make based on the environment for targets in a working
     directory
     """
-    maxThreads = 2 + multiprocessing.cpu_count()
     command = ["cmake", "--build", wd, f"-j{maxThreads}"]
     if len(targets) > 0:
         command += ["--target"] + targets
     subprocess.check_call(command, cwd=wd)
 
-def compileTargets(targets):
+def compileTargets(targets, maxThreads):
     """
-    Compile given targets. Be as parallel as possible
+    Compile given targets. Unless specified, be as parallel as possible
     """
     for s in configuredSuites():
         sTargets = [x.target for x in targets if x.suite == s]
         if len(sTargets) == 0:
             continue
         print(f"Compiling targets from suite {s}: {', '.join(sTargets)}")
         try:
-            invokeBuild(os.path.join(buildDir, s), sTargets)
+            invokeBuild(os.path.join(buildDir, s), sTargets, maxThreads)
         except Exception as e:
             sys.exit(f"Build of target {', '.join(sTargets)} failed with {e}. See output above\n")
 
-def compileSuite(suite):
+def compileSuite(suite, maxThreads):
     print(f"Compiling suite {suite}")
     try:
-        invokeBuild(os.path.join(buildDir, suite), [])
+        invokeBuild(os.path.join(buildDir, suite), [], maxThreads)
     except Exception as e:
         sys.exit(f"Build of suite {suite} failed with {e}. See output above\n")
 
-def cleanSuite(suite):
+def cleanSuite(suite, maxThreads):
```

Comment:
I feel like the cleaning should not use -j since it just calls the OS command for remove - and doesn't really use CPU power

Reply:
or at least not by default